diff -ur uClibc-nps-orig/ldso/ldso/arc/dl-sysdep.h uClibc-nps-devel/ldso/ldso/arc/dl-sysdep.h
--- uClibc-nps-orig/ldso/ldso/arc/dl-sysdep.h	2016-08-18 17:23:31.849210462 +0100
+++ uClibc-nps-devel/ldso/ldso/arc/dl-sysdep.h	2016-08-18 17:22:52.228323075 +0100
@@ -138,7 +138,6 @@
 /* Return the run-time load address of the shared object.  */
 static __always_inline Elf32_Addr elf_machine_load_address(void)
 {
-#ifdef __UCLIBC_HAS_THREADS_NATIVE__
     /* To find the loadaddr we subtract the runtime addr of a non-local symbol
      * say _DYNAMIC from it's build-time addr.
      * N.B., gotpc loads get optimized by the linker if it finds the symbol
@@ -156,15 +155,6 @@
         "sub %0, %0, %1                ;delta"                    "\n"
         : "=&r" (addr), "=r"(tmp)
     );
-#else
-	Elf32_Addr addr, tmp;
-	__asm__ (
-        "ld  %1, [pcl, _dl_start@gotpc] ;build addr of _dl_start   \n"
-        "add %0, pcl, _dl_start@pcl ;runtime addr of _dl_start \n"
-        "sub %0, %0, %1                 ;delta                     \n"
-         : "=&r" (addr), "=r"(tmp)
-     );
-#endif
 	return addr;
 }
 
